src/lib/prisma-session-store.ts

```diff
@@ -1,440 +1,467 @@
+      await this.prisma.session.create({
+        data: { ...data, data: sessionString },
+      });
     }
 
-    /**
-     * @description A function to generate the Prisma Record ID for a given session ID
-     *
-     * Note: If undefined and dbRecordIdIsSessionId is also undefined then a random
-     * CUID will be used instead.
-     */
-    private readonly dbRecordIdFunction = (sid: string) =>
-      this.options.dbRecordIdFunction?.(sid) ?? cuid();
-
-    /**
-     * Disables store, used when prisma cannot be connected to
-     */
-    private disable(): void {
-      this.invalidConnection = true;
-    }
+    if (callback) defer(callback);
+  };
 
-    /**
-     * Returns if the connect is valid or not, logging an error if it is not.
-     */
-    private async validateConnection(): Promise<boolean> {
-      await (
-        this.prisma?.$connect?.() ??
-        Promise.reject(new Error('Could not connect'))
-      ).catch(() => {
-        this.disable();
-        this.stopInterval();
-        this.logger.error(dedent`Could not connect to Sessions model in Prisma.
-        Please make sure that prisma is setup correctly and that your migrations are current.
-        For more information check out https://github.com/kleydon/prisma-session-store`);
+  /**
+   * A function to stop any ongoing intervals and disconnect from the `PrismaClient`
+   */
+  public async shutdown(): Promise<void> {
+    this.stopInterval();
+    await this.prisma.$disconnect();
+  }
+
+  /**
+   * Start an interval to prune expired sessions
+   */
+  public startInterval(): void {
+    const ms = this.options.checkPeriod;
+
+    if (typeof ms === 'number' && ms !== 0) {
+      this.stopInterval();
+      this.checkInterval = setInterval(() => {
+        this.prune();
+      }, Math.floor(ms));
+    }
+  }
+
+  /**
+   * Stop checking if sessions have expired
+   */
+  public stopInterval(): void {
+    if (this.checkInterval) clearInterval(this.checkInterval);
+  }
+
+  /**
+   * Refresh the time-to-live for the session with the given `sid`.
+   *
+   * @param sid the id of the session to refresh
+   * @param session the data of the session to resave
+   * @param callback a callback notifying that the refresh was completed
+   * or that an error occurred
+   */
+  public readonly touch = async (
+    sid: string,
+    session: PartialDeep<Express.SessionData>,
+    callback?: (err?: unknown) => void
+  ) => {
+    if (!(await this.validateConnection())) return callback?.();
+
+    const ttl = getTTL(this.options, session, sid);
+    const expires = createExpiration(ttl);
+
+    try {
+      const existingSession = await this.prisma.session.findOne({
+        where: { sid },
       });
 
-      return !this.invalidConnection;
+      if (existingSession !== null) {
+        const existingSessionData = {
+          ...this.serializer.parse(existingSession.data ?? '{}'),
+          cookie: session.cookie,
+        };
+
+        await this.prisma.session.update({
```

Comment:
Does does `prisma.session.update()` _require_ `data: {...}` to include _all_ fields, or just the field(s) to be updated? 
(Wondering if perhaps only the `expires` value needs to be included here, for the update to be effective...)

Reply:
`.update()` only requires the changed fields.
However the `data` is also required as `touch()` takes in a `session` that is used to update the `cookie` with the new expiration to expire the cookie on the client side as well.

Reply:
Ah - ok.